### DISCUSSION TITLE
Using Coral M.2 in Home Assistant OS

### DISCUSSION THREAD

**Discussion by normalee1993** [blakeblackshear/frigate#10220] [General]
Hi there,

I am using a coral m.2 TPU in my wifi slot and it had worked previously during a proof of concept phase with Home Assistant OS and Frigate add-on and it is no longer work in my released system.

-------How do I correct my configuration to enable the Croal M.2 EdgeTPU?--------

—Here is my error in addon logs of HomeAssistant:—
2024-03-03 23:28:28.212141230 [2024-03-03 17:28:28] frigate.detectors.plugins.edgetpu_tfl INFO : Attempting to load TPU as /dev/apex/apex_0
2024-03-03 23:28:28.212164628 Process detector:coral_pci:
2024-03-03 23:28:28.212222809 [2024-03-03 17:28:28] frigate.detectors.plugins.edgetpu_tfl ERROR : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.

—Coral TPU info from HA Hardware Info:—
DEVNAME: /dev/apex_0
DEVPATH: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
MAJOR: ‘120’
MINOR: ‘0’
SUBSYSTEM: apex

—Current and not working FRIGATE configuration:—
detectors:
coral_pci:
type: edgetpu
device: /dev/apex/apex_0

Different iterations of Frigate detector section with no success:
detectors:
coral:
type: edgetpu
device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral:
type: edgetpu
device: devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral1:
type: edgetpu
device: pci:0

detectors:
coral1:
type: edgetpu
device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral1:
type: edgetpu
device: /dev/apex_0

detectors:
coral_pci:
type: edgetpu
device: /dev/apex_0

detectors:
coral_pci:
type: edgetpu
device: pci:0

detectors:
coral_pci:
type: edgetpu
device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral_pci:
type: edgetpu
device: /dev/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral_pci:
type: edgetpu
device: /pci0000:00/0000:00:01.3/0000:02:00.0

detectors:
coral_pci:
type: edgetpu
device: /pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0

detectors:
coral_pci:
type: edgetpu
device: /apex/apex_0

detectors:
coral_pci:
type: edgetpu
device: /dev/apex/apex_0

**Comment by SloPOS** [comment]
Same issue here; shows as apex0 in HAOS, but won't pass through for some reason to frigate.

**Reply by SloPOS** [reply]
I was able to fix mine by installing Frigate (Full Access) and disabling protection mode. Then this config worked:

detectors:
  coral:
    type: edgetpu
    device: pci


As an added bonus this enabled my AMD Graphics also that had been giving me some trouble.

**Comment by eloijrseganfredo** [comment]
> Hi there,
> 
> I am using a coral m.2 TPU in my wifi slot and it had worked previously during a proof of concept phase with Home Assistant OS and Frigate add-on and it is no longer work in my released system.
> 
> -------How do I correct my configuration to enable the Croal M.2 EdgeTPU?--------
> 
> —Here is my error in addon logs of HomeAssistant:— 2024-03-03 23:28:28.212141230 [2024-03-03 17:28:28] frigate.detectors.plugins.edgetpu_tfl INFO : Attempting to load TPU as /dev/apex/apex_0 2024-03-03 23:28:28.212164628 Process detector:coral_pci: 2024-03-03 23:28:28.212222809 [2024-03-03 17:28:28] frigate.detectors.plugins.edgetpu_tfl ERROR : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.
> 
> —Coral TPU info from HA Hardware Info:— DEVNAME: /dev/apex_0 DEVPATH: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0 MAJOR: ‘120’ MINOR: ‘0’ SUBSYSTEM: apex
> 
> —Current and not working FRIGATE configuration:— detectors: coral_pci: type: edgetpu device: /dev/apex/apex_0
> 
> Different iterations of Frigate detector section with no success: detectors: coral: type: edgetpu device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral: type: edgetpu device: devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral1: type: edgetpu device: pci:0
> 
> detectors: coral1: type: edgetpu device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral1: type: edgetpu device: /dev/apex_0
> 
> detectors: coral_pci: type: edgetpu device: /dev/apex_0
> 
> detectors: coral_pci: type: edgetpu device: pci:0
> 
> detectors: coral_pci: type: edgetpu device: /devices/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral_pci: type: edgetpu device: /dev/pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral_pci: type: edgetpu device: /pci0000:00/0000:00:01.3/0000:02:00.0
> 
> detectors: coral_pci: type: edgetpu device: /pci0000:00/0000:00:01.3/0000:02:00.0/apex/apex_0
> 
> detectors: coral_pci: type: edgetpu device: /apex/apex_0
> 
> detectors: coral_pci: type: edgetpu device: /dev/apex/apex_0

I'm in the same situation, maybe someone managed to solve this?
